examples/components/CCFRAUD/training/run.py

```diff
@@ -0,0 +1,440 @@
+                    loss = self.criterion_(predictions, labels.type(torch.float))
+                    if net_loss is not None:
+                        loss += net_loss * 1e-5
+
+                    # Compute gradients and adjust learning weights
+                    loss.backward()
+                    self.optimizer_.step()
+
+                    precision, recall = precision_recall(
+                        preds=predictions.detach(), target=labels
+                    )
+                    train_metrics.add_metric("precision", precision.item())
+                    train_metrics.add_metric("recall", recall.item())
+                    train_metrics.add_metric(
+                        "accuracy",
+                        accuracy(
+                            preds=predictions.detach(), target=labels, threshold=0.5
+                        ).item(),
+                    )
+                    train_metrics.add_metric(
+                        "loss", (loss.detach() / data.shape[0]).item()
+                    )
+                    train_metrics.step()
+
+                    if (i + 1) % num_of_batches_before_logging == 0 or (i + 1) == len(
+                        self.train_loader_
+                    ):
+                        log_message = [
+                            f"Epoch: {epoch}/{self._epochs}",
+                            f"Iteration: {i+1}/{len(self.train_loader_)}",
+                        ]
+
+                        for name, value in train_metrics.get_step().items():
+                            log_message.append(f"{name}: {value}")
+                            self.log_metrics(
+                                mlflow_client,
+                                root_run_id,
+                                name,
+                                value,
+                            )
+                        logger.info(", ".join(log_message))
+                        train_metrics.reset_step()
+
+                test_metrics = self.test()
+
+                log_message = [
+                    f"Epoch: {epoch}/{self._epochs}",
+                ]
+
+                # log test metrics after each epoch
+                for name, value in test_metrics.get_global().items():
+                    log_message.append(f"{name}: {value}")
+                    self.log_metrics(
+                        mlflow_client,
+                        root_run_id,
+                        name,
+                        value,
+                    )
+                logger.info(", ".join(log_message))
+
+                # log metrics at the pipeline level
+                for name, value in train_metrics.get_global().items():
+                    log_message.append(f"{name}: {value}")
+                    self.log_metrics(
+                        mlflow_client,
+                        root_run_id,
+                        name,
+                        value,
+                        pipeline_level=True,
+                    )
```

Comment:
this metric seems to be a bit off. Can you pass `pipeline_level=False` if you are logging on each epoch? Also, let's log one more train and test metric for each FL iteration (`pipeline_level=True`). You can take reference from the MNIST example.

Reply:
Hi @garg-amit, passing `pipeline_level=True` is in this case intentional as this should be the logging the metrics for whole epoch and I expected that this is the correct way. I believe that as a user you may want to have an overview of progress within nodes for each epoch at pipeline level. What are your thoughts on that?

Reply:
Right, @majercakdavid! We definitely need to provide a way for users to look at metrics for each epoch at the pipeline level but all the metrics by default show the metrics at the pipeline level rn. What I meant with `pipeline_level=True` param is to report metrics for each iteration (along with metrics for each epoch). I know the param name is a bit confusing- it should be named as `iteration_level`. Happy to elaborate on it on Teams if it's still confusing!

Reply:
Hi @garg-amit please can you confirm that the updated solution is what you expected and resolve the conversation? Thank you :)